elastic/kibana
# [Maps] fix map app crashing on file upload request timeout

Fixes https://github.com/elastic/kibana/issues/95744

There are 2 problems occurring in https://github.com/elastic/kibana/issues/95744
1) poor network connection is causing timeouts when uploading geojson files
2) bug in file upload UI component causes page to break on http errors.

PR adds a "Upload file in smaller chunks" switch. When true, file is uploaded in 100KB chunks instead of 1MB chunks. This works well for files with small to medium features. This does not resolve the problem for files with large features as there is no way to chunk upload size smaller than a single feature. Resolving this problem for files with large features is more complicated (either feature simplification for dropping large features) and is considered out of scope for this PR.
<img width="300" alt="Screen Shot 2022-09-12 at 12 29 16 PM" src="https://user-images.githubusercontent.com/373691/189730314-37413ba0-3a9e-4b99-80f8-616a43da5d08.png">


PR fixes the bug in the file upload UI component so http errors are properly displayed and the page does not break
<img width="300" alt="Screen Shot 2022-08-30 at 8 53 20 AM" src="https://user-images.githubusercontent.com/373691/187521796-4df61c14-b4f7-43ed-8097-0a37fc2934c5.png">

To test:
1) download [administrative regions geojson](https://maps.elastic.co/#file/administrative_regions_lvl2)
2) create new map, click "Add layer", select "Upload file".
3) Open browser tools and throttle network speed to "fast 3g"
4) select administrative regions geojson, Set "Upload size" to "small", and click "Import file".
5) It takes a long time but file should import.
